src/exla/exla_nif_util.cc

```diff
@@ -0,0 +1,42 @@
+#include "exla/exla_nif_util.h"
+
+int ExlaNifUtil::get(ErlNifEnv* env, ERL_NIF_TERM term, int &var){ return enif_get_int(env, term, &var); }
+int ExlaNifUtil::get(ErlNifEnv* env, ERL_NIF_TERM term, long int &var){ return enif_get_int64(env, term, &var); }
+int ExlaNifUtil::get(ErlNifEnv* env, ERL_NIF_TERM term, long long int &var){ return enif_get_int64(env, term, (long int*) &var); }
+
+int ExlaNifUtil::get(ErlNifEnv* env, ERL_NIF_TERM term, std::string &var){
```

Comment:
I am a bit torn about this one. Are we trying to convert either a charlist or a binary into a C++ string? What is the use case for passing any of them?

Similarly to the atoms stuff, I would prefer to ensure only a certain type of them is given to C++ string and any conversion or normalize happens in the Elixir entrypoint. :)